app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -4,21 +4,13 @@ class TwoFactorAuthenticationController < ApplicationController
 
     before_action :check_remember_device_preference
 
-    # rubocop:disable Metrics/MethodLength
     def show
-      if current_user.piv_cac_enabled?
-        redirect_to login_two_factor_piv_cac_url
-      elsif current_user.totp_enabled?
-        redirect_to login_two_factor_authenticator_url
-      elsif phone_enabled?
-        validate_otp_delivery_preference_and_send_code
-      else
-        redirect_to two_factor_options_url
-      end
+      return if redirect_on_non_phone_enabled
```

Comment:
Style: another option would be
```ruby
redirect_on_non_phone_enabled || redirect_on_phone_enabled || redirect_to two_factor_options_url
```

or

```ruby
redirect_to two_factor_url

...

def two_factor_url
  if current_user.piv_cac_enabled?
    login_two_factor_piv_cac_url
  elsif current_user.webauthn_configurations.any?
    login_two_factor_webauthn_url
  elsif ...
  else
    two_factor_options_url
  end
end
```

This would require a tweak to `validate_otp_delivery_preference_and_send_code` so it returns a url.